### ISSUE TITLE
`stack path` is slower than usual

### ISSUE THREAD

**Issue by fredefox** [commercialhaskell/stack#3446]
### General summary/comments (optional)

Running `stack path --compiler-bin` prompts stack to start "Populating index cache" this is new behavior and didn't happen before. The issue for me is that I use this command in my shells login-script and it slows down initialization considerably.

My issue could be worked around if the stack installation maintained setting global environment variables.

I also do realize that this is perhaps not a standard use-case, however, I would like to have the programs that stack installs (`ghc` etc.) globally available and this is how I did this.

### Steps to reproduce

For example:

1. Log in to a new shell.
2. Run command `stack path --compiler-bin` or simply `stack path`.

### Expected

The path is printed immediately.

### Actual

It prints

"Populating index cache ... <delete line character / CR?>
Populated index cache."

### Stack version

```
$ stack --version
Version 1.5.1, Git revision 600c1f01435a10d127938709556c1682ecfd694e (4861 commits) x86_64 hpack-0.17.1
```

### Method of installation

I believe I used the one-liner from this site https://docs.haskellstack.org/en/stable/install_and_upgrade/

**Comment by mgsloan** [comment]
Makes sense, shouldn't need to populate the index in order to do `stack path`.

**Comment by andrewcchen** [comment]
`stack path` is also slow for me, but I don't see "Populating index cache".

When I run `stack path -v`, I can see that it hangs on decoding build plan for more than 1 second.
```
2017-09-23 16:03:28.548437: [debug] Decoding build plan from: /home/andrew/.stack/build-plan/lts-9.5.yaml
@(Stack/Snapshot.hs:151:5)                                                                                                              
2017-09-23 16:03:29.722151: [debug] Run process: /sbin/ldconfig -p
@(System/Process/Log.hs:37:3)                                                                                                           
```

**Comment by ivan-m** [comment]
I also commented on this in #3423

**Comment by fredefox** [comment]
This behavior is back with:

    Version 1.4.0, Git revision 5986498a9d1073e8d58c79ec1302bb2a7847a396 (4642 commits) x86_64 hpack-0.17.0

**Comment by danbroooks** [comment]
I have done a PR that implements a speed improvement for this command: https://github.com/commercialhaskell/stack/pull/5645 I just thought I would mention here that when implementing that PR, on `v2.7.3`  I do not see this `Populating index cache` message. It might be that this is no longer an issue with stack?
